### PR TITLE
Add EXACT CARDINALITY commands

### PR DESCRIPTION
**Pull request by e-dard**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated

This PR introduces five new commands:

```
SHOW SERIES EXACT CARDINALITY [ON] [FROM] [WHERE] [GROUP BY] [LIMIT] [OFFSET]
SHOW MEASUREMENT EXACT CARDINALITY [ON] [FROM] [WHERE] [GROUP BY] [LIMIT] [OFFSET]
SHOW FIELD KEY EXACT CARDINALITY [ON] [FROM] [WHERE] [GROUP BY] [LIMIT] [OFFSET]
SHOW TAG KEY EXACT CARDINALITY [ON] [FROM] [WHERE] [GROUP BY] [LIMIT] [OFFSET]
SHOW TAG VALUES EXACT CARDINALITY [ON] [FROM] WITH KEY = "x" [WHERE] [GROUP BY] [LIMIT] [OFFSET]
```

Further, two versions of both the already-implemented `SHOW SERIES CARDINALITY` and `SHOW MEASUREMENT CARDINALITY` commands have been converted to utilise the HLL++ sketches that are maintained in the index. These estimations are [very accurate](https://github.com/influxdata/influxdb/pull/8512#issuecomment-313709923), and very fast.

Specifically, the variants that currently support cardinality estimation are

```
SHOW SERIES CARDINALITY;
SHOW SERIES CARDINALITY ON db0;
SHOW MEASUREMENT CARDINALITY
SHOW MEASUREMENT CARDINALITY ON db0;
```

All other `SHOW X CARDINALITY` commands will continue to use the exact-counting approaches of the `SHOW X EXACT CARDINALITY` commands. They will be converted to estimations in future releases.

### CODE REVIEW

File: coordinator/statement_executor.go [hunk 1]
```diff
@@ -703,6 +707,18 @@ func (e *StatementExecutor) executeShowMeasurementsStatement(q *influxql.ShowMea
...
 }
 
+func (e *StatementExecutor) executeShowMeasurementCardinalityStatement(stmt *influxql.ShowMeasurementCardinalityStatement) (models.Rows, error) {
+	n, err := e.TSDBStore.MeasurementsCardinality(stmt.Database)
+	if err != nil {
+		return nil, err
+	}
+
+	return []*models.Row{&models.Row{
+		Columns: []string{"cardinality"},
```
Comment (e-dard) [inline coordinator/statement_executor.go]: @benbjohnson @jsternberg question over what your thoughts are on this name?

File: coordinator/statement_executor.go [hunk 2]
```diff
@@ -758,6 +774,19 @@ func (e *StatementExecutor) executeShowShardsStatement(stmt *influxql.ShowShards
 	return rows, nil
 }
 
+func (e *StatementExecutor) executeShowSeriesCardinalityStatement(stmt *influxql.ShowSeriesCardinalityStatement) (models.Rows, error) {
+	n, err := e.TSDBStore.SeriesCardinality(stmt.Database)
+	if err != nil {
+		return nil, err
+	}
+
+	return []*models.Row{&models.Row{
```
Comment (e-dard) [inline coordinator/statement_executor.go]: Eurgh I realise this doesn't match the measurements case... Let's come to an agreement about what we want to output and I'll fix up the branch to do that for both commands.

File: coordinator/statement_executor.go (hunk 1 above)
Comment (benbjohnson) [reply coordinator/statement_executor.go]: I think `cardinality` works for me. 👍

**Comment by e-dard** [comment]
@jsternberg changes made. Thanks.

**Pull request merged**